### PR TITLE
support soft wallets

### PR DESCRIPTION
**Pull request by willscott**
wraps the eth.accounts on confidential.accounts so that calls to `signTransaction` get intercepted and encrypted.

I'll add todo's for doing the same with the signing of call's / other data, which is routed through the account provider in a separate channel.

### CODE REVIEW

**Comment by willscott** [comment]
fix #121

File: test/softwallet.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      from: acct.address,
+      gas: '0x3b9aca00',
+      gasPrice: '0x3b9aca00'
+    });
+    if (!contract) {
+      throw new Error('contract should not be empty');
+    }
+    let mthd = contract.methods.incrementCounter();
+    await mthd.send({
+      gas: '0x3b9aca00',
```
Comment (armaniferrante) [inline test/softwallet.js]: Do we have to specify the gas here for any reason? In the past it was symptomatic of a bug.

Comment (willscott) [reply test/softwallet.js]: the accounts signer will make a request out to the mock gateway to estimate gas if this isn't specified. this wrapping should cause that to happen after encryption, so it'll get the correct gas for the confidential operation. This was easier that plumbing through the extra method in the mock gateway.

Comment (armaniferrante) [reply test/softwallet.js]: 👍 sounds reasonable.

**Pull request merged**